src/core/pdfFiller/fillPdf.ts

```diff
@@ -19,8 +20,24 @@ export function fillPDF(pdf: PDFDocument, fieldValues: Field[]): PDFDocument {
         `Field ${index}, ${pdfField.getName()} expected ${expected}`
       )
     }
-
-    if (pdfField instanceof PDFCheckBox) {
+    // First handle radio groups. If the value for this field
+    // is a RadioSelect object, then assume the pdfField
+    // has children, and check the correct box given the index value.
+    // Idea taken from this comment:
+    // https://github.com/Hopding/pdf-lib/issues/780#issuecomment-771453157
+    if (_.isObject(value)) {
+      const children = pdfField.acroField.getWidgets()
+      const setValue = children[value.select].getOnValue()
+      if (setValue !== undefined) {
+        pdfField.acroField.dict.set(PDFName.of('V'), setValue)
+        children[value.select].setAppearanceState(setValue)
+      } else {
+        console.error(children)
+        throw new Error(
+          `Error handling RadioGroup, could not set index ${value.select}`
+        )
+      }
```

Comment:
What is the advantage of this over checking and filling pdf radio groups using the radio group api directly? For example this is what is being used for the OR-40 pdf radio groups

```
else if (pdfField instanceof PDFRadioGroup) {
   try {
        pdfField.select(value?.toString() || '')
```

Reply:
We need more tooling around testing buggy pdfs, but the pdffield in the IL form—even though it behaves as a radio group with clicks—is of type PDFCheckbox. I eventually figured out that the widget itself has children. But iterating over the list of fields would never detect them. 

So the advantage is now we have two ways to select radio buttons depending on how they were implemented by the poor person that has to make these forms.

Reply:
I will update the comments to indicate it’s for cases where the widget is implemented this way. I hadn’t come across a field passing instanceof PDFRadioGroup yet.

Reply:
Interesting, it's a bummer the radiogroup isn't used in the IL form

Reply:
I don't know pdf well enough. Somehow the field behaves as a radio group but for some reason pdflib doesn't detect it as such. But that link in my comment shows other people have run into this too.